tracing/tracing.go

```diff
@@ -5,46 +5,37 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
-	"strconv"
 
 	jaegercfg "github.com/uber/jaeger-client-go/config"
 )
 
-// New registers Jaeger as the OpenTracing implementation.
-// If jaegerAgentHost is an empty string, tracing is disabled.
-func New(agentHost, serviceName, samplerType string, samplerParam float64) io.Closer {
-	if agentHost != "" {
-		cfg := jaegercfg.Configuration{
-			Sampler: &jaegercfg.SamplerConfig{
-				SamplingServerURL: fmt.Sprintf("http://%s:5778/sampling", agentHost),
-				Type:              samplerType,
-				Param:             samplerParam,
-			},
-			Reporter: &jaegercfg.ReporterConfig{
-				LocalAgentHostPort: fmt.Sprintf("%s:6831", agentHost),
-			},
-		}
-
-		closer, err := cfg.InitGlobalTracer(serviceName)
-		if err != nil {
-			fmt.Printf("Could not initialize jaeger tracer: %s\n", err.Error())
-			os.Exit(1)
-		}
-		return closer
+// InstallJaeger registers Jaeger as the OpenTracing implementation.
+func InstallJaeger(serviceName string, cfg *jaegercfg.Configuration) io.Closer {
+	closer, err := cfg.InitGlobalTracer(serviceName)
+	if err != nil {
+		fmt.Printf("Could not initialize jaeger tracer: %s\n", err.Error())
+		os.Exit(1)
 	}
-	return ioutil.NopCloser(nil)
+	return closer
 }
 
 // NewFromEnv is a convenience function to allow tracing configuration
 // via environment variables
+// Tracing is disabled unless one of the following environment variables is used to configure jaeger:
```

Comment:
So here we say unless one of the following... but below on line 35 we check that all 3 must not be empty. I think this needs rephrasing to match the code.

Reply:
🤦‍♂️ good catch

Reply:
Actually I think they do match up. I'll reword the comment to cancel out some negatives.